File: 0520-detect-capital/0520-detect-capital.java
class Solution {
    public boolean detectCapitalUse(String word) {
        int upperCnt = 0;

        for (char c : word.toCharArray()) {
            if (Character.isUpperCase(c)) upperCnt++;
        }

        return upperCnt == 0 ||
               upperCnt == word.length() ||
               (upperCnt == 1 && Character.isUpperCase(word.charAt(0)));
    }
}

File: 0520-detect-capital/0520-detect-capital.kt
class Solution {
    fun detectCapitalUse(word: String): Boolean {
        val upperCnt = word.count { it.isUpperCase() }
        return upperCnt == 0 ||
               upperCnt == word.length ||
               (upperCnt == 1 && word[0].isUpperCase())
    }
}